src/limb.rs

```diff
@@ -110,31 +146,70 @@ impl <'a> Range<'a> {
                 let mut dest_as_bytes = limbs_as_bytes_mut(dest);
                 try!(rng.fill(&mut dest_as_bytes));
             }
-            dest[self.max_exclusive.len() - 1] &= most_significant_limb_mask;
+            dest[self.max_exclusive.len() - 1] &= self.sampling_most_sig_limb_mask;
 
             if self.are_limbs_within(&dest) {
                 return Ok(());
             }
+
+            if self.sampling_reduce_when_over_bound {
```

Comment:
I worry breaking the series of `if self.are_limbs_within ... return Ok` checks into separate functions would make the pattern harder to follow, and I'm not sure how much sense tests of this part in isolation would make, but i'll poke at it.

Regardless, the indentation on the comment is definitely annoying. I wonder if it makes sense to move most of it into a sort of footnote?